Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean

```diff
@@ -355,10 +463,6 @@ instance instIsScalarTowerCoordinateRing : IsScalarTower R R[X] W.CoordinateRing
   Quotient.isScalarTower R R[X] _
 #align weierstrass_curve.coordinate_ring.is_scalar_tower WeierstrassCurve.Affine.CoordinateRing.instIsScalarTowerCoordinateRing
 
-instance instSubsingletonCoordinateRing [Subsingleton R] : Subsingleton W.CoordinateRing :=
-  Module.subsingleton R[X] _
-#align weierstrass_curve.coordinate_ring.subsingleton WeierstrassCurve.Affine.CoordinateRing.instSubsingletonCoordinateRing
-
```

Comment:
I think the following section organisation is better. Move the entire `Algebra` section "The coordinate ring as an `R[X]`-algebra" (from `instAlgebraCoordinateRing` to `smul_basis_mul_Y`) to right before the `Ring` section "Ideals in the coordinate ring over a ring". This won't literally compile because of misplaced definitions, but the following changes should make it work: (1) move `quotientXYIdealEquiv` from the `Algebra` section to right after `XYIdeal_add_eq` in the `Ring` section because it needs the definition of `XYIdeal`; (2) move `mk` from the `Ring` section to right before `basis` in the `Algebra` section because some of the lemmas for `basis` require this definition.